### ISSUE TITLE
MPI improve performance on PPO?

### ISSUE THREAD

**Issue by 1576012404** [openai/baselines#941]
i want to know how to use MPI to improve the performance of training when i have enough environements in each process. It seems that MPI can not speed up training comparing without running MPI. 
i train Pong with ppo . num_env=32 nstep=128 notepochs=8 nminibatches=8

**Comment by DanielTakeshi** [comment]
@1576012404 You need to be way more specific with your question for us to give advice. What timing benchmarks do you have, what are you hoping to get, etc?

Keep in mind that PPO by default uses vectorized environments which already have parallelism.

**Comment by 1576012404** [comment]
#filename:train.py
from baselines.ppo2 import ppo2
from baselines.common.vec_env.subproc_vec_env import SubprocVecEnv
from baselines.common.atari_wrappers import make_atari, wrap_deepmind

def EnvFunc(iSeed):
    def InnerFunc():
        env_id = "PongNoFrameskip-v4"
        oEnv = make_atari(env_id)
        oEnv.seed(iSeed)
        oEnv=wrap_deepmind(oEnv,frame_stack=True)
        return oEnv
    return InnerFunc

learning_rate=5e-4
clip_range=0.1
n_timesteps=int(1e7)
hyperparmas ={'nsteps': 128, 'noptepochs': 8, 'nminibatches': 8, 'lr': learning_rate, 'cliprange':clip_range ,
            'vf_coef': 0.5, 'ent_coef': 0.01}
def Train():
    num_env = 32
    env = SubprocVecEnv([EnvFunc(i) for i in range(num_env)])
    act = ppo2.learn(
        network="cnn",
        env=env,
        total_timesteps=n_timesteps,
        **hyperparmas,
    )


if __name__=="__main__":
    Train()



here is the code. mpirun -np 2 python train.py or run python train.py directly ,the steps each of mpi process need to train are almost the same as run directly. that means  when i run with mpi i need twice of times.so mpi can not speed up training?
